ml-agents/mlagents/trainers/tests/test_environments/test_simple.py

```diff
@@ -0,0 +1,256 @@
+    Very simple "game" - the agent has a position on [-1, 1], gets a reward of 1 if it reaches 1, and a reward of -1 if
+    it reaches -1. The position is incremented by the action amount (clamped to [-step_size, step_size]).
+    """
+
+    def __init__(self):
+        self._brains: Dict[str, BrainParameters] = {}
+        self._brains[BRAIN_NAME] = BrainParameters(
+            brain_name=BRAIN_NAME,
+            vector_observation_space_size=OBS_SIZE,
+            num_stacked_vector_observations=1,
+            camera_resolutions=[],
+            vector_action_space_size=[1],
+            vector_action_descriptions=["moveDirection"],
+            # vector_action_space_type=0,  # "discrete"
+            vector_action_space_type=1,  # "continuous"
+        )
+
+        # state
+        self.position = 0.0
+        self.step_count = 0
+
+    def step(
+        self,
+        vector_action: Dict[str, Any] = None,
+        memory: Dict[str, Any] = None,
+        text_action: Dict[str, Any] = None,
+        value: Dict[str, Any] = None,
+    ) -> AllBrainInfo:
+        if vector_action:
+            delta = vector_action[BRAIN_NAME][0][0]
+            self.position += clamp(delta, -STEP_SIZE, STEP_SIZE)
+            self.position = clamp(self.position, -1, 1)
+            self.step_count += 1
+        done = self.position >= 1.0 or self.position <= -1.0
+        reward = self.position if done else 0.0
+
+        agent_info = AgentInfoProto(
+            stacked_vector_observation=[self.position] * OBS_SIZE,
+            reward=reward,
+            done=done,
+        )
+        return {
+            BRAIN_NAME: BrainInfo.from_agent_proto(
+                0, [agent_info], self._brains[BRAIN_NAME]
+            )
+        }
+
+    def reset(
+        self,
+        config: Dict[str, float] = None,
+        train_mode: bool = True,
+        custom_reset_parameters: Any = None,
+    ) -> AllBrainInfo:  # type: ignore
+        self.position = 0.0
+        self.step_count = 0
+        agent_info = AgentInfoProto(
+            stacked_vector_observation=[self.position] * OBS_SIZE,
+            done=False,
+            max_step_reached=False,
+        )
+        return {
+            BRAIN_NAME: BrainInfo.from_agent_proto(
+                0, [agent_info], self._brains[BRAIN_NAME]
+            )
+        }
+
+    @property
+    def global_done(self):
+
+        return self.step_count > 1000
+
+    @property
+    def external_brains(self) -> Dict[str, BrainParameters]:
+        return self._brains
+
+    @property
+    def reset_parameters(self) -> Dict[str, str]:
+        return {}
+
+    def close(self):
+        pass
+
+
+class EnvContext:
+    def __init__(self, env: BaseUnityEnvironment):
+        self.env = env
+        self.previous_step: StepInfo = StepInfo(None, {}, None)
+        self.previous_all_action_info: Dict[str, ActionInfo] = {}
+
+
+# Copied from SubprocessEnvManager and removed the subprocess part
+class LocalEnvManager(EnvManager):
```

Comment:
@harperj here's the un-subprocess'ed version of SubprocessEnvManager. I can simplify it a bit further to only ever handle a single environment.

Reply:
If you're generally happy with the PR, I'll either move this to ml-agents-envs, or a different file in this test directory (depending on whether we think it's generally useful).